qa/L0_stability_perf/check_results.py

```diff
@@ -0,0 +1,123 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import sys
+import glob
+import os
+from collections import defaultdict
+import re
+
+
+class TestOutputValidator:
+    """
+    Functions that validate the output
+    of the test
+    """
+    def __init__(self, test_name, threshold):
+        self._tolerance_percent = threshold
+
+        check_function = self.__getattribute__(f'check_{test_name}')
+
+        if check_function():
+            sys.exit(0)
+        else:
+            sys.exit(1)
+
+    def check_perf_stability(self):
+        """
+        Get the perf results from each model analyzer run
+        as well as perf_analyzer run and compare them
+        """
+
+        log_paths = glob.glob(os.path.join(os.getcwd(), f"*.log"))
+
+        # Split the logs up into log pairs model_name: [model_name.test.log, model_name.perf.log]
+        # Note that the values may be ordered differently (perf log first then test log)
+        log_pairs = defaultdict(list)
+        for log_path in log_paths:
+            model_name = os.path.relpath(log_path, os.getcwd()).split('.')[0]
+            log_pairs[model_name].append(log_path)
+
+        # compare the values within paired logs
+        for model_name, log_pair in log_pairs.items():
+            throughputs, latencies = [], []
+            for log in sorted(log_pair):
+                # Open log and match regex
+                with open(log, 'r+') as f:
+                    log_contents = f.read()
+
+                throughput = re.search('Throughput: (\d+\.\d+|\d+)',
+                                       log_contents)
+                if throughput:
+                    throughputs.append(float(throughput.group(1)))
+                else:
+                    print(f"\n***\n*** No throughput found in {log} \n***")
+                    return False
+
+                p99_latency = re.search('p99 latency: (\d+\.\d+|\d+)',
+                                        log_contents)
+                if p99_latency:
+                    latencies.append(float(p99_latency.group(1)))
+                else:
+                    print(f"\n***\n*** No throughput found in {log} \n***")
+                    return False
+
+            # Once all throughputs and latencies are collected, compute diff percentage
+            throughput_diff = 100 * (throughputs[0] -
+                                     throughputs[1]) / throughputs[0]
+
+            latency_diff = 100 * (latencies[1] - latencies[0]) / latencies[1]
+
+            if throughput_diff > self._tolerance_percent and latency_diff > self._tolerance_percent:
+                print(
+                    f"\n***\n*** Model Analyzer throughput and latency differ "
+                    f"greatly from perf analyzer standalone for model {model_name}. "
+                    f"\n*** perf analyzer values: Throughput={throughputs[0]} infer/sec, p99 Latency={latencies[0]} usec. \n***"
```

Comment:
Maybe include the MA results here as well?

Reply:
The MA results are appended on error